nx-X11/lib/configure.ac

```diff
@@ -0,0 +1,283 @@
+    freebsd*)
+        XTHREAD_CFLAGS="-D_THREAD_SAFE"
+        XTHREADLIB="-pthread" ;;
+    dragonfly*|openbsd*)
+        XTHREADLIB="-pthread" ;;
+    solaris*)
+        XTHREAD_CFLAGS="-D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS" ;;
+    esac
+fi
+AC_SUBST(XTHREADLIB)
+AC_SUBST(XTHREAD_CFLAGS)
+
+AC_CHECK_FUNC(poll, [AC_DEFINE(USE_POLL, 1, [poll() function is available])], )
+
+#
+# Find keysymdef.h
+#
+AC_MSG_CHECKING([keysym definitions])
+AC_ARG_WITH([keysymdefdir],
+        [AC_HELP_STRING([--with-keysymdefdir=DIR], [The location of keysymdef.h (defaults to xproto include dir)])],
+        [KEYSYMDEFDIR=$withval],
+        [KEYSYMDEFDIR=`pwd`/../include])
+
+if test ! -d "$KEYSYMDEFDIR"; then
+        AC_MSG_ERROR([$KEYSYMDEFDIR doesn't exist or isn't a directory])
+fi
+
+FILES="keysymdef.h XF86keysym.h Sunkeysym.h DECkeysym.h HPkeysym.h"
+for i in $FILES; do
+    if test -f "$KEYSYMDEFDIR/$i"; then
+            KEYSYMDEFS="$KEYSYMDEFS $KEYSYMDEFDIR/$i"
+    elif test "x$i" = "xkeysymdef.h"; then
+            AC_MSG_ERROR([Cannot find keysymdef.h])
+    fi
+done
+AC_MSG_RESULT([$KEYSYMDEFS])
+AC_SUBST(KEYSYMDEFS)
+
+AC_ARG_ENABLE(xcms,
+              AS_HELP_STRING([--disable-xcms],
+                [Disable Xlib support for CMS *EXPERIMENTAL*]),
+              [XCMS=$enableval],[XCMS=yes])
+AM_CONDITIONAL(XCMS, [test x$XCMS = xyes ])
+if test x"$XCMS" = "xyes"; then
+        AC_DEFINE(XCMS,1,[Include support for XCMS])
+fi
+
+AC_ARG_ENABLE(xlocale,
+              AS_HELP_STRING([--disable-xlocale],
+                [Disable Xlib locale implementation *EXPERIMENTAL*]),
+              [XLOCALE=$enableval],[XLOCALE=yes])
+
+AM_CONDITIONAL(XLOCALE, [ test x$XLOCALE = xyes ])
+if test x"$XLOCALE" = "xyes"; then
+        AC_DEFINE(XLOCALE,1,[support for X Locales])
+fi
+
+
+# This disables XLOCALEDIR.  Set it if you're using BuildLoadableXlibI18n,
+# don't have either issetugid() or getresuid(), and you need to protect
+# clients that are setgid or setuid to an id other than 0.
+AC_MSG_CHECKING([if XLOCALEDIR support should be enabled])
+AC_ARG_ENABLE(xlocaledir,
+              AS_HELP_STRING([--enable-xlocaledir],
+                [Enable XLOCALEDIR environment variable support]),
+              [ENABLE_XLOCALEDIR=$enableval],[ENABLE_XLOCALEDIR=$XLOCALEDIR_IS_SAFE])
+if test "x$ENABLE_XLOCALEDIR" = "xno"; then
+        AC_DEFINE(NO_XLOCALEDIR,1,[Disable XLOCALEDIR environment variable])
+fi
+AC_MSG_RESULT($ENABLE_XLOCALEDIR)
+
+AC_ARG_ENABLE(xf86bigfont,
+              AS_HELP_STRING([--disable-xf86bigfont],
+                [Disable XF86BigFont extension support]),
+              [XF86BIGFONT=$enableval],[XF86BIGFONT="yes"])
+if test "x$XF86BIGFONT" = "xyes"; then
+# FIXME: Skip check for proto headers for now... Requires .pc files for all the proto
+# header packages we have. Keeping that in mind for later...
+#    PKG_CHECK_MODULES(BIGFONT, [xf86bigfontproto >= 1.2.0],
+        AC_DEFINE(XF86BIGFONT,1,[Enable XF86BIGFONT extension])
```

Comment:
> Besides that, a general question: aren't we running into general problems here? First, it's not about the system-side X.Org proto headers but our nx-specific ones, right? Also, that would essentially make those a build dependency, which makes bootstrapping libNX_X11 difficult, since we'd need to make sure that proto packages are available before building libNX_X11. It even gets more complicated since in our case everything is built from one source package and if any component fails nothing will be available. That's definitely a delicate matter.

I am still dreaming of a time where we build against latest X.org. That's why I leave those references to the x11 proto headers in, although we ship the proto headers in nx-libs ourselves for the time being.

Doing nothing about this now. It is for later.